src/library/scala/collection/Iterable.scala

```diff
@@ -713,6 +713,24 @@ object IterableOps {
 @SerialVersionUID(3L)
 object Iterable extends IterableFactory.Delegate[Iterable](immutable.Iterable) {
   implicit def toLazyZipOps[A, CC[X] <: Iterable[X]](that: CC[A]): LazyZipOps[A, CC[A]] = new LazyZipOps(that)
+
+  /**
+    * @return an Iterable by using a function `f` producing elements of
+    *         type `A` and updating an internal state `S`.
+    * @param init State initial value
+    * @param f    Computes the next element (or returns `None` to signal
+    *             the end of the collection)
+    * @tparam A   Type of the elements
+    * @tparam S   Type of the internal state
+    */
+  def unfold[A, S](init: S)(f: S => Option[(A, S)]): Iterable[A] = new UnfoldIterable(init)(f)
+
+  /** An `Iterable` that calls [[scala.collection.Iterator.unfold `Iterator.unfold`]]
+    * to create `Iterator`s.
+    */
+  private final class UnfoldIterable[A, S](initial: S)(f: S => Option[(A, S)]) extends AbstractIterable[A] {
+    override def iterator: Iterator[A] = Iterator.unfold(initial)(f)
+  }
```

Comment:
UnfoldIterable doesn't memoize items since it creates new UnfoldIterator every time iteration is done.

Reply:
I should probably document `Iterable.unfold` better to indicate that it doesn't memoize elements

Reply:
I've checked scala.collection.View and it seems a better fit than scala.collection.Iterable to put collections without memoization. Views are not memoized by default, while Iterables are backed by immutable.List by default. Instead of changing the semantics for one method (unfold) in Iterable companion object, we can move that method (unfold) to View companion object. WDYT? @LPTK

Reply:
It shouldn't be difficult to add it to all of the factories - although that would still end up with a memoizing `View`. I'm honestly not sure what the best course of action is.

Reply:
If we add a `View.Unfold` class, then we can have `Factory.from(new View.Unfold(initial)(f))` for all types basically for free